mindsdb/interfaces/agents/agents_controller.py

```diff
@@ -311,15 +313,37 @@ def get_completion(
         Raises:
             ValueError: Agent's model does not exist.
         '''
+        # Log the human question
+        human_question = "\n".join(f"{key}: {value}" for message in messages for key, value in message.items())
+        self.save_agent_history(
+            agent_id=agent.id,
+            type='question',
+            text=human_question,
+            sender_type='Human',
+            trace_id=trace_id,
+            observation_id=observation_id,
+            tools=None,
+            stream_id=None
+        )
+
         if stream:
-            return self._get_completion_stream(
+            response = self._get_completion_stream(
                 agent,
                 messages,
                 trace_id=trace_id,
                 observation_id=observation_id,
                 project_name=project_name,
                 tools=tools
             )
+            return self.log_from_iterator(
+                iterator=response,
+                agent_id=agent.id,
+                type='completion',
+                sender_type='AI',
+                trace_id=trace_id,
+                observation_id=observation_id,
+                stream_id=str(uuid.uuid4())
```

Comment:
I think during stream observation_id or trace_id (or both) will be equal and this `stream_id` adds no new information